client/a8c-for-agencies/sections/marketplace/issue-license/licenses-form/index.tsx

```diff
@@ -0,0 +1,301 @@
+							return { ...product, quantity };
+						}
+
+						return item;
+					} )
+				);
+
+				// Unselecting the current selected variant
+				dispatch(
+					recordTracksEvent( 'calypso_a4a_marketplace_issue_license_unselect_product', {
+						product: replace.slug,
+						quantity,
+					} )
+				);
+
+				dispatch(
+					recordTracksEvent( 'calypso_a4a_marketplace_issue_license_select_product', {
+						product: product.slug,
+						quantity,
+					} )
+				);
+			} else {
+				handleSelectBundleLicense( product );
+			}
+		},
+		[ dispatch, handleSelectBundleLicense, quantity, selectedLicenses, setSelectedLicenses ]
+	);
+
+	const isReady = true;
+
+	const isSelected = useCallback(
+		( slug: string | string[] ) =>
+			selectedLicenses.some(
+				( license ) =>
+					( Array.isArray( slug ) ? slug.includes( license.slug ) : license.slug === slug ) &&
+					license.quantity === quantity
+			),
+		[ quantity, selectedLicenses ]
+	);
+
+	const onClickVariantOption = useCallback(
+		( product: APIProductFamilyProduct ) => {
+			dispatch(
+				recordTracksEvent( 'calypso_a4a_marketplace_issue_license_variant_option_click', {
+					product: product.slug,
+				} )
+			);
+		},
+		[ dispatch ]
+	);
+
+	const isSingleLicenseView = quantity === 1;
+
+	const getProductCards = ( products: APIProductFamilyProduct[] ) => {
+		return products.map( ( productOption ) =>
+			Array.isArray( productOption ) ? (
+				<LicenseMultiProductCard
+					key={ productOption.map( ( { slug } ) => slug ).join( ',' ) }
+					products={ productOption }
+					onSelectProduct={ onSelectOrReplaceProduct }
+					onVariantChange={ onClickVariantOption }
+					isSelected={ isSelected( productOption.map( ( { slug } ) => slug ) ) }
+					selectedOption={ productOption.find( ( option ) =>
+						selectedLicenses.find(
+							( license ) => license.slug === option.slug && license.quantity === quantity
+						)
+					) }
+					isDisabled={
+						! isReady ||
+						( isIncompatibleProduct( productOption, incompatibleProducts ) &&
+							! isSelected( productOption.map( ( { slug } ) => slug ) ) )
+					}
+					hideDiscount={ isSingleLicenseView }
+					suggestedProduct={ suggestedProduct }
+					quantity={ quantity }
+				/>
+			) : (
+				<LicenseProductCard
+					isMultiSelect
+					key={ productOption.slug }
+					product={ productOption }
+					onSelectProduct={ onSelectProduct }
+					isSelected={ isSelected( productOption.slug ) }
+					isDisabled={ ! isReady || isIncompatibleProduct( productOption, incompatibleProducts ) }
+					hideDiscount={ isSingleLicenseView }
+					suggestedProduct={ suggestedProduct }
+					quantity={ quantity }
+				/>
+			)
+		);
+	};
+
+	if ( isLoadingProducts ) {
+		return (
+			<div className="licenses-form">
+				<div className="licenses-form__placeholder" />
+			</div>
+		);
+	}
+
+	return (
+		<div className="licenses-form">
+			<QueryProductsList type="jetpack" currency="USD" />
```

Comment:
I don't think it matters now, but we can remove the `type` prop here so it loads all products instead of Jetpack only.